Structurization and formatting of code

mealet opened the pull request:
## Description
I've changed some structure's ideas and namings for clearer understand. Also `Mozilla` style switched to `Google` with expanded config in `.clang-format`. Google format style looks more readable and easier to interact with API.
Besides that makefile was changed to emit all C files in `src` directory and its subs. Also binary file compilation (or _TARGET_) moved to `build` directory.

I only did some formattings and configurations, main logics isn't touched.
But some moments might be changed in future (especially memory management).
Is this code, or any pieces of it AI generated?

Full changelog below.

## Changelog
1. Re-configured tools configs such as: `.clangd`, `.clang-format`, `.clang-tidy`
2. Edited makefile to emit all C files
3. Formatting style changed from `Mozilla` to `Google`
4. Formatted code (added extra lines, clearer variables namings)
5. Constants renamed with Google Style Standards (camelCase with `k` at start)
6. Changed namings in header files: `PND_MNAME_H_` (where `MNAME` - module name).

jx853 commented on src/parse.c:
when you move a file use git mv

jx853 commented on src/parse.h:
same thing with this file

jx853 reviewed (changes requested):
this looks good except for the issue of not using git mv. We will handle memory management once everything is working.

mealet commented:
I've rebased commits and added `git mv`, but because of changes git added deletions.
Also I've changed namings in header files: `PND_MNAME_H_` (where `MNAME` - module name).
Changelog updated
